### PR TITLE
Denormalize course_id into annotations

### LINKED ISSUE #2529: Polling annotations is a slow query
The last few days we can notice a slight increase in database load. When looking at `show full processlist;` on our database server, the queries selecting unanswered questions pops up a lot:
```sql
SELECT `annotations`.* FROM `annotations`
   INNER JOIN `submissions` 
     ON `annotations`.`submission_id` = `submissions`.`id` 
   WHERE `annotations`.`type` = 'Question' 
     AND `submissions`.`course_id` = 123 
     AND `annotations`.`question_state` = 2 
    ORDER BY `annotations`.`updated_at` DESC
    LIMIT 10
    OFFSET 0;
```

- We should add an index to make this query faster (since this query is performed a lot).
- The `'type' = 'Question'` comparison feels like it is slow because it compares strings.
- Maybe we can think about caching this page?

### PR DESCRIPTION
**Pull request by chvp**
Also add index to annotations for common queries.

Closes #2529.

### CODE REVIEW

**Review by bmesuere — APPROVED** [review]

File: app/models/course.rb [hunk 1]
```diff
@@ -126,27 +126,28 @@ class Course < ApplicationRecord
...
+           inverse_of: :course
 
   has_many :answered_questions,
            lambda {
              where question_state: :answered
            },
-           through: :submissions,
-           source: :questions
+           class_name: "Question",
+           inverse_of: :course
```
Comment (chvp) [inline app/models/course.rb:150]: These changes make sure the field is used for the query.

File: db/migrate/20210308102217_add_index_to_annotation.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class AddIndexToAnnotation < ActiveRecord::Migration[6.1]
+  def change
+    add_column :annotations, :course_id, :integer
```
Comment (niknetniko) [inline db/migrate/20210308102217_add_index_to_annotation.rb:3]: Is there a downside to also making this a foreign key?

File: app/models/annotation.rb [hunk 3]
```diff
@@ -6,17 +6,19 @@
...
 #  evaluation_id      :bigint
 #  type               :string(255)      default("Annotation"), not null
 #  question_state     :integer
 #  last_updated_by_id :integer          not null
+#  course_id          :integer          not null
 #
 class Annotation < ApplicationRecord
   include ApplicationHelper
 
+  belongs_to :course
```
Comment (niknetniko) [inline app/models/annotation.rb:21]: Should this be `belongs_to :course, optional: true` if the course is deleted?

File: db/migrate/20210308102217_add_index_to_annotation.rb (hunk 2 above)
Comment (chvp) [reply db/migrate/20210308102217_add_index_to_annotation.rb:3]: No, forgot about that, my bad.

**Review by bmesuere — APPROVED** [review]

**Review by niknetniko — APPROVED** [review]

**Pull request merged**